lib/stemcell/launcher.rb

```diff
@@ -184,17 +196,18 @@ def launch(opts={})
 
       # generate user data script to bootstrap instance, include in launch
       # options UNLESS we have manually set the user-data (ie. for ec2admin)
-      launch_options[:user_data] = opts.fetch('user_data', render_template(opts))
+      launch_options[:user_data] = Base64.encode64(opts.fetch('user_data', render_template(opts)))
 
+      # add tags to launch options so we don't need to make a separate CreateTags call
+      launch_options[:tag_specifications] = [{
+        resource_type: 'instance',
+        tags: tags.map { |k, v| { key: k, value: v } }
+      }]
       # launch instances
```

Comment:
Since we're not doing any post launch tasks like set tags or classic link association -- this requested change is to set termination protection at launch, correct?

Reply:
Yes. Set at launch time for anything that can be set at launch time.